src/stores/user-store.js

```diff
@@ -38,6 +38,8 @@ class UserStore extends EventEmitter {
         const index = this._users.findIndex((oldUser) => user.id === oldUser.id);
         if (index) {
             this._users[index] = user;
+        } else {
```

Comment:
if we can't find the user with the `id`, the `findIndex` method returns `-1` and the line ` this._users[index] = user;` will be executed any way